### ISSUE TITLE
Wrong indication of "Shorthand struct initialization"

### ISSUE THREAD

**Issue by ibhagwan** [rust-lang/rust-analyzer#13502]
**rust-analyzer version**: `rust-analyzer 0.3.1258-standalone`

**rustc version**: `rustc 1.64.0 (a55dd71d5 2022-09-19)`

The following code generates an incorrect warning:
```rust
struct Foo {
    pub bar: i8
}

impl Foo {
    pub fn new() -> std::io::Result<Foo> {
        let bar: i8 = 1;   // change this variable name to get rid of the warning
        Ok(Foo {
            bar: bar       // this line generates a "Shorthand struct initialization" warning
        })
    }
}
```

Changing the local variable to `baz` gets rid of the warning, it seems `rust_analyzer` gets confused by the local variable name being identical to the struct field name.

**Comment by bjorn3** [comment]
The hint is correct. `Foo { bar: bar }` can be written as `Foo { bar }`. And this is in fact what the associated quickfix will turn your code into.

**Comment by ibhagwan** [comment]
> The hint is correct. `Foo { bar: bar }` can be written as `Foo { bar }`. And this is in fact what the associated quickfix will turn your code into.

I see, I guess I misunderstood the issue, ty!
